core/src/providers/openai.rs

```diff
@@ -688,37 +689,35 @@ pub async fn embed(
 
 pub struct OpenAILLM {
     id: String,
+    host: Option<String>,
+    use_eu_endpoint: bool,
     api_key: Option<String>,
 }
 
 impl OpenAILLM {
     pub fn new(id: String) -> Self {
-        OpenAILLM { id, api_key: None }
-    }
-
-    #[inline]
-    fn host(use_openai_eu_host: bool) -> &'static str {
-        if use_openai_eu_host {
-            "eu.api.openai.com"
-        } else {
-            "api.openai.com"
+        OpenAILLM {
+            id,
+            host: None,
+            use_eu_endpoint: false,
```

Comment:
Do we use this ? I'm confused because it looks we're getting the info from `credentials`

Reply:
It comes from credentials and we just store it in the object to use later, just like we store `self.api_key`.